pkg/spec/job_test.go

```diff
@@ -127,3 +127,130 @@ func TestJobSpecMod_CreateKubernetesController(t *testing.T) {
 		})
 	}
 }
+
+func TestJobValidate(t *testing.T) {
+	tests := []struct {
+		name    string
+		input   *JobSpecMod
+		success bool
+	}{
+		{
+			name: "Set restart policy as failure",
+			input: &JobSpecMod{
+				ControllerFields: ControllerFields{
+					ObjectMeta: meta_v1.ObjectMeta{
+						Name: "testJob",
+					},
+					Controller: "job",
+					PodSpecMod: PodSpecMod{
+						PodSpec: api_v1.PodSpec{
+							Containers: []api_v1.Container{
+								{
+									Name:  "testContainer",
+									Image: "testImage",
+								},
+							},
+							RestartPolicy: api_v1.RestartPolicyAlways,
+						},
+					},
+				},
+			},
+			success: false,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			if err := test.input.Validate(); err != nil && test.success == true {
+				// test failing condition
+				t.Fatalf("test expected to pass, but failed with error: %v", err)
+			} else if err != nil && test.success == false {
+				// test passing condition
+				t.Logf("test expected to fail, failed with error: %v", err)
+			} else if err == nil && test.success == true {
+				// test passing condition
+				t.Logf("test passed")
+			} else if err == nil && test.success == false {
+				// test failing condition
+				t.Fatalf("test expected to fail, but passed with error")
+			}
+		})
+	}
+}
+
+func TestJobFix(t *testing.T) {
+	tests := []struct {
+		name    string
+		input   *JobSpecMod
+		output  *JobSpecMod
+		success bool
+	}{
+		{
+			name: "no restartPolicy given",
```

Comment:
there should be at least one test case that is testing when fix is returning an error (`success=false`)

Reply:
done!